Branch-SDK/src/main/java/io/branch/referral/util/BranchEvent.java

```diff
@@ -256,6 +269,175 @@ public boolean logEvent(Context context) {
         return isReqQueued;
     }
 
+    /**
+     * Logs a Branch Commerce Event based on an in-app purchase
+     *
+     * @param context  Current context
+     * @param purchase Respective purchase
+     */
+    public void logEventWithPurchase(Context context, @NonNull Purchase purchase) {
+        List<String> productIds = purchase.getProducts();
+        List<QueryProductDetailsParams.Product> productList = new ArrayList<>();
+        List<QueryProductDetailsParams.Product> subsList = new ArrayList<>();
+
+        for (String productId : productIds) {
+            QueryProductDetailsParams.Product inAppProduct = QueryProductDetailsParams.Product.newBuilder()
+                    .setProductId(productId)
+                    .setProductType(BillingClient.ProductType.INAPP)
+                    .build();
+
+            QueryProductDetailsParams.Product subsProduct = QueryProductDetailsParams.Product.newBuilder()
+                    .setProductId(productId)
+                    .setProductType(BillingClient.ProductType.SUBS)
+                    .build();
+
+            productList.add(inAppProduct);
+            subsList.add(subsProduct);
+        }
+
+        QueryProductDetailsParams queryProductDetailsParams =
+                QueryProductDetailsParams.newBuilder()
+                        .setProductList(productList)
+                        .build();
+
+        QueryProductDetailsParams querySubsProductDetailsParams =
+                QueryProductDetailsParams.newBuilder()
+                        .setProductList(subsList)
+                        .build();
+
+        PurchasesUpdatedListener purchasesUpdatedListener = new PurchasesUpdatedListener() {
+            @Override
+            public void onPurchasesUpdated(BillingResult billingResults, List<Purchase> purchases) {
+
+            }
+        };
+
+        BillingClient billingClient = BillingClient.newBuilder(context)
```

Comment:
You're creating a second BillingClient instance here, is that recommended?

From the doc

```
It's strongly recommended that you instantiate only one BillingClient instance at one time to avoid multiple [PurchasesUpdatedListener.onPurchasesUpdated(BillingResult, List)](https://developer.android.com/reference/com/android/billingclient/api/PurchasesUpdatedListener#onPurchasesUpdated(com.android.billingclient.api.BillingResult,%20java.util.List%3Ccom.android.billingclient.api.Purchase%3E)) callbacks for a single event.
```

https://developer.android.com/reference/com/android/billingclient/api/BillingClient